specs/dotsigstore_bundle/verifier.go

```diff
@@ -0,0 +1,115 @@
+// Pseudo-code for an example verification flow for a sigstore bundle,
+// with some input parameters required from the client.
+
+type Verifier interface{
+	// Verify verifier the provided data against the signatre,
+	// optionally relying on the kid as hint to which key to use.
+	Verify(data, sig []byte, kid string) error
+}
+
+
+type Options {
+	// Verifier for user provided keys. May know about many keys.
+	Verifier Verifier
+	// Optional user provided Root CA
+	RootCA *x509.Certificate
+	// Optional Rekor URL
+	RekorURL string
+	// Optional Fulcio URL
+	FulcioURL string
+}
+
+// Verify is the main verirication function.
+// It accepts a required sigstore bundle, and an optional path to the blob.
+// In if the attestation in the bundle is a DSSE envelope, the blob path
+// is ignored.
+func Verify(b *bundle, blob string, opts Options) error {
+	if b.AttestationDsse != nil {
+		if err = verifyDsse(b, opts); err != nil {
+			return err
+		}
+	} else {
+		if err = verifyBlob(b, blob, opts); err != nil {
+			return err
+		}
+	}
+
+	if b.rekorEntry != nil {
```

Comment:
is it correct that you must also enter this path if/when a certificate (if used) is expired?

Reply:
likewise, does the final cert validation wrt to timestamp need to occur here or in the outer loop?

Reply:
This will be addressed in an updated coming soon, have it locally but need to run it through once more before pushing. Will add explicit logic for cert validation wrt to timestamp in `verifyRekor`.